_posts/2020-04-13-deprecation-warnings-rails-guide.markdown

```diff
@@ -0,0 +1,107 @@
+---
+layout: post
+title: "The Complete Guide for Deprecation Warnings in Rails"
+date: 2020-04-13 12:00:00
+categories: ["rails", "upgrades"]
+author: luciano
+---
+
+Deprecation warnings are a common thing in our industry. They are warnings that notify us that a specific feature (e.g. a method) will be removed soon (usually in the next [minor or major](https://semver.org/) version) and should be replaced with something else.
+Features are deprecated rather than immediately removed, in order to provide backward compatibility (a solution that works in both the current and the future version), and to give programmers time to implement the code in a way that follows the new standard.
+
+In this guide we'll show you what's the workflow that we use at [FastRuby.io](https://www.fastruby.io) to address deprecation warnings when we [upgrade Rails applications](https://www.fastruby.io/blog/tags/upgrades).
+
+<!--more-->
+
+## Finding
+
+There are a few different ways to find deprecation warnings in your application.
+
+If you have good test coverage, you can run the whole test suite and look at the logs that were generated. If you are using a CI service (like [CircleCI](https://circleci.com/) or [Travis CI](https://travis-ci.org/)) you can easily see the logs once the build finished running. Otherwise, if you run the tests locally, you can look at the output on the console or in the `log/test.log` file.
```

Comment:
```suggestion
If you have good test coverage, you can run the whole test suite and look at the logs that were generated. If you are using a CI service (like [CircleCI](https://circleci.com/) or [Travis CI](https://travis-ci.org/)) you can easily see the logs once the build finishes running. Otherwise, if you run the tests locally, you can look at the output in the console or in the `log/test.log` file.
```